targets/unity-v2/make.js

```diff
@@ -394,7 +438,7 @@ function getRequestActions(tabbing, apiCall) {
         return tabbing + "request.TitleId = request.TitleId ?? PlayFabSettings.TitleId;\n";
     if (apiCall.auth === "SessionTicket")
         return tabbing + "if (!IsClientLoggedIn()) throw new PlayFabException(PlayFabExceptionCode.NotLoggedIn,\"Must be logged in to call this method\");\n";
-    if (apiCall.auth === "SecretKey")
+    if (apiCall.auth === "SecretKey" && isApiInstance === false)
```

Comment:
>if (apiCall.auth === "SecretKey" && isApiInstance === false) [](start = 4, length = 60)

We need to add correct login here when isApiInstance === true. Please take a look here how it is done for C#:

[https://github.com/PlayFab/SDKGenerator/blob/master/targets/csharp/make.js#L300](https://github.com/PlayFab/SDKGenerator/blob/master/targets/csharp/make.js#L300)

Please add that.

Reply:
We check this control in PlayFabHttp.MakeApiCall method. you said me that to move that there. So I think we should not add isApiInstance === true, we should remove  isApiInstance === false condition also. What is our final decision what should I do?

Reply:
I removed it for now.  After your review if you want I can add it again.

Reply:
Ok, my bad. I see that some of the logic that i wanted is indeed already moved to MakeApiCall. Please see my comment there.

---
In reply to: [262365219](https://github.com/PlayFab/SDKGenerator/pull/420#discussion_r262365219) [](ancestors = 262365219)